### PR TITLE
[Bug Fix] PBI-9,10,11: Create, See, and Edit Announcement

### PR DESCRIPTION
**Pull request by fxdros**
Bug fix for infinite rendering

### CODE REVIEW

File: src/resources/app/pages/course/Announcement.vue [hunk 1]
```diff
@@ -10,6 +10,11 @@ import AppLayout from "../../components/Layout/AppLayout";
 
 export default {
   layout: [AppLayout, CourseStream],
-  components: {AnnouncementContainer}
+  components: {AnnouncementContainer},
+  beforeCreate() {
+    if (!this.$store.state.isAuthenticated) {
+      window.location.assign("/login");
+    }
+  }
```
Comment (IlmaMannix) [inline src/resources/app/pages/course/Announcement.vue:18]: I think you should add some props in this page, so you could fetch param 'id' from URL

File: src/resources/app/components/Container/AnnouncementContainer.vue [hunk 2]
```diff
@@ -1,14 +1,15 @@
 <template>
   <div id="container">
     <CreateAnnouncement
-      :getAnnouncement="this.getAnnouncements()"
+      :get-announcement="this.getAnnouncements"
```
Comment (IlmaMannix) [inline src/resources/app/components/Container/AnnouncementContainer.vue:4]: camel case please 🥺🥺

File: src/resources/app/components/AnnouncementCard.vue [hunk 3]
```diff
@@ -6,10 +6,11 @@
   >
     <div id="announcement">
       <AnnouncementContent
-        :name=this.$props.createdBy
+        :name=this.$props.author
```
Comment (IlmaMannix) [inline src/resources/app/components/AnnouncementCard.vue:9]: Just wanna ask, what's the difference between this.$props.author and "this.$props.author" ?

File: src/resources/app/components/Container/AnnouncementContainer.vue [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
     />
     <div v-for="announcement in announcements" v-bind:key="announcement.id">
       <AnnouncementCard
         :content="announcement.content"
-        :created-by="announcement.author"
+        :author="announcement.author"
         :date="announcement.createdAt"
         :id="announcement.id"
+        :author-id="announcement.createdBy"
```
Comment (IlmaMannix) [inline src/resources/app/components/Container/AnnouncementContainer.vue:12]: camel case please 🥺🥺

**Review by IlmaMannix — APPROVED** [review]

File: src/resources/app/components/Container/AnnouncementContainer.vue (hunk 2 above)
Comment (nofamex) [inline src/resources/app/components/Container/AnnouncementContainer.vue:4]: yep definitely camel case

**Review by nofamex — APPROVED** [review]
overall pretty good to me, gj mate

**Pull request merged**